src/libs/elektra/keyname.c

```diff
@@ -1824,6 +1824,57 @@ ssize_t keySetNamespace (Key * key, elektraNamespace ns)
 	return key->keySize;
 }
 
+/**
+ * Checks if the given key name part is an array part.
+ *
+ * The return value of this function can safely be treated as a boolean.
+ *
+ * @param namePart an arbitrary string that shall be checked
+ *
+ * @returns if @p namePart is an array part, the index of the first digit in @p namePart,
+ *          or 0 otherwise
+ */
+int elektraIsArrayPart (const char * namePart)
```

Comment:
Should be in a static helper lib?

Reply:
Maybe but let's avoid such discussions for now. We can decide that when we implement #4245.